lm75bd/lm75bd.c

```diff
@@ -27,7 +27,23 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
+  uint8_t buf[2];
+
+  uint8_t addr = 0x0U;
+  i2cSendTo(devAddr, &addr, 1);
+
+  i2cReceiveFrom(devAddr, buf, sizeof(buf));
+
+  uint8_t b1 = buf[0];
+  uint8_t b2 = buf[1];
+
+  uint16_t result = ((b1 << 8) | b2) >> 5 ;
   
+     if (result & 0x0400) { 
+        result = ~(result | 0xF800) + 1;
+        *temp = result * -0.125;
+     } else *temp = result * 0.125;
```

Comment:
This code is unnecessary and can be supplied to using a type of `int16_t` for the result variable instead of `uint16_t` and then multiplying by 0.125